lsstseries/ensemble.py

```diff
@@ -199,6 +205,17 @@ def _build_index(self, obj_id, band):
         # Create a multiindex
 
         idx = range(len(list(zip(obj_id, band))))
+
+        #idx to count up for each unique pair of obj_id,band 
+        count_dict = {}
```

Comment:
I got stuck here, so I'm going to try to unpack my understanding and see if it matches the intent here.

You have might have (simplified) input like:
```
obj_id = [1,2,2,2,2,2,1]
band = [u,u,g,u,g,g,g]
```
Where you need those objects and bands to be the same length. you zip that up and get
`[1u,2u,2g,2u,2g,2g,1g]`

you want to end up with some nice indexed table that looks like
`[1u1,2u1,2g1,2u2,2g2,2g3,1g1]`

Do you need to maintain the order of the inputs? Because you could otherwise do something cute with

`obj_bands, counts = np.unique(zip(obj_id, band), axis=0, return_counts=true)`

That I think might read a little easier?

Either way, I think it would be helpful to have some comments here describing what your input expectations are, what mutations you're performing, and what the output should look like. Unit tests would also help document the expectations.

Reply:
I think you have the right idea, though I'm running into an error getting your toy example to work. 
-> `AxisError: axis 0 is out of bounds for array of dimension 0`

Preserving the order of inputs is not a requirement at this point, but an improvement to this would be to order by timestamp most likely.

I'll link the issue here which has an example of what smart indexing wants to achieve input/output wise #10

I'll also mention here, that the motivation for this smart indexing is subtly influenced by a potential refactor to Xarray, and as such the same caveat for dropna applies here, this could be dropped if we go in the direction of Vaex or Dask.Dataframe as they don't support multi-indices.

Reply:
I think it would be helpful to include the expected input/output alongside the method, either in docstring comments or unit tests that exercise the various options (preferably the unit tests).

Reply:
A unit test is a good idea, I'll write one up later today

Reply:
Unit test is in!